presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -37,6 +37,18 @@ public static String arrayAverage()
                 "s -> if(s[2] = 0, cast(null as double), s[1] / cast(s[2] as double)))";
     }
 
+    @SqlInvokedScalarFunction(value = "array_split", deterministic = true, calledOnNullInput = false)
+    @Description("Returns an array of arrays splitting input array into slices of given length. Ignores null elements.")
+    @TypeParameter("T")
+    @SqlParameters({@SqlParameter(name = "input", type = "array(T)"), @SqlParameter(name = "sz", type = "int")})
+    @SqlType("array(array(T))")
+    public static String arraySplit()
+    {
+        return "RETURN transform(" +
+                "sequence(1, cardinality(input), sz), " +
+                "x -> slice(input, x, sz))";
```

Comment:
In Presto we have a limit of 10000 elements for SEQUENCE function. So will be good to check and fail:

RETURN IF(cardinality(input)/sz > 10000, fail('Cannot split array of size: ' || cast(cardinality(input) as varchar))', transfrorm...)